### PR TITLE
Cache GoroutineInfo result

### PR DESCRIPTION
**Pull request by icholy**
#234

### CODE REVIEW

File: proc/proc.go [hunk 1]
```diff
@@ -446,6 +447,11 @@ func (dbp *Process) SwitchGoroutine(gid int) error {
 // Returns an array of G structures representing the information
 // Delve cares about from the internal runtime G structure.
 func (dbp *Process) GoroutinesInfo() ([]*G, error) {
+
```
Comment (derekparker) [inline proc/proc.go]: Kill this empty line.

File: proc/proc.go [hunk 2]
```diff
@@ -39,6 +39,7 @@ type Process struct {
 	// Normally SelectedGoroutine is CurrentThread.GetG, it will not be only if SwitchGoroutine is called with a goroutine that isn't attached to a thread
 	SelectedGoroutine *G
 
+	gCache                  []*G
```
Comment (derekparker) [inline proc/proc.go]: Rename to `AllGCache`.

**Comment by derekparker** [comment]
Let's move the cache clearing to `dbp.run`. That function wraps any function which continues the process. Not all functions that continue the process end up calling `dbp.Halt`. This way, any time the process executes we clear this cache.

File: proc/proc.go (hunk 2 above)
Comment (icholy) [inline proc/proc.go]: @derekparker  Exported?

**Comment by icholy** [comment]
@derekparker done & rebased

**Comment by derekparker** [comment]
@icholy sorry, I mistyped before: let's not export the cache member. Change to `allGCache` and I'll merge.

**Comment by icholy** [comment]
@derekparker  done

**Comment by icholy** [comment]
@derekparker want me to update the commit message style?

**Comment by derekparker** [comment]
@icholy yes, please. Thanks.

**Comment by icholy** [comment]
@derekparker done

**Comment by derekparker** [comment]
Can you add `Fixes #<issue no>` to the end of the commit. Other than that, looks good.

**Comment by icholy** [comment]
@derekparker updated

**Comment by derekparker** [comment]
Merged, thanks!

**Pull request closed without merging**